book_source/03_intermediate_users_guide/01_pecan_xml.Rmd

```diff
@@ -1,195 +1,487 @@
-* **update** : [optional] Should previous results of meta.analysis and get.traits be re-used. If set to TRUE the meta-analysis and get.trait.data will always be executed. Setting this to FALSE will try and reuse existing results. Future versions will allow for AUTO as well which will try and reuse if the PFT/traits have not changed. The default value is FALSE.
-* **threshold** threshold for Gelman-Rubin convergence diagnostic (MGPRF); default is 1.2
+* `name`: [optional] name of host server where model is located and executed, if not specified localhost is assumed.
+* `rundir`: [optional/required] location where all the configuration files are written. For localhost this is optional (`<outdir>/run` is the default), for any other host this is required.
+* `outdir`: [optional/required] location where all the outputs of the model are written. For localhost this is optional (`<outdir>/out` is the default), for any other host this is required.
+* `scratchdir`: [optional] location where output is written. If specified the output from the model is written to this folder and copied to the outdir when the model is finished, this could significantly speed up the model execution (by using local or ram disk).
+* `clearscratch`: [optional] if set to TRUE the scratchfolder is cleaned up after copying the results to the outdir, otherwise the folder will be left. The default is to clean up after copying.
+* `qsub`: [optional] the command to submit a job to the queuing system. There are 3 parameters you can use when specifying the qsub command, you can add additional values for your specific setup (for example -l walltime to specify the walltime, etc). You can specify @NAME@ the pretty name, @STDOUT@ where to write stdout and @STDERR@, where to write stderr. You can specify an empty element (`<qsub/>`) in which case it will use the default value is `qsub -V -N @NAME@ -o @STDOUT@ -e @STDERR@ -s /bin/bash`.
+* `qsub.jobid`: [optional] the regular expression used to find the `jobid` returned from `qsub`. If not specified (and `qsub` is) it will use the default value is `Your job ([0-9]+) .*`
+* `qstat`: [optional] the command to execute to check if a job is finished, this should return DONE if the job is finished. There is one parameter this command should take `@JOBID@` which is the ID of the job as returned by `qsub.jobid`. If not specified (and qsub is) it will use the default value is `qstat -j @JOBID@ || echo DONE`
+* `job.sh`: [optional] additional options to add to the job.sh at the top.
 
+## Advanced features {#xml-advanced}
 
-## Ensemble Runs
+### `ensemble`: Ensemble Runs {#xml-ensemble}
 
-Only if this section is defined an ensemble analysis is done.
+As with `meta.analysis`, if this section is missing, then PEcAn will not do an ensemble analysis.
```

Comment:
important question: can this section actually be missing, or do you always have to specify an ensemble with at least one member? If it is missing, does PEcAn just add it and default to n=1?